src/java.base/share/classes/jdk/internal/classfile/impl/BufWriterImpl.java

```diff
@@ -111,6 +111,47 @@ public void writeU2(int x) {
         this.offset = offset + 2;
     }
 
+    @ForceInline
+    public void writeU2(int x1, int x2) {
```

Comment:
Perhaps a more descriptive name would be `writeU1U1` here, then `writeU1U2`, `writeU1U1U1` and `writeU2U2` for the next methods, respectively? Either way these methods will be a bit of an eye-sore, but let's at least iron out any ambiguities.